### PR TITLE
Split tests into internal and external

### PR DESCRIPTION
**Pull request by phadej**
- external test uses only exposed API

Also

- move library under src/ (using `.` is bad practice)
- remove -fno-ignore-asserts from tests, they did nothing (this is not transitive flag)

### CODE REVIEW

**Review by jorisdral — APPROVED** [review]
I don't mind the restructuring. @dcoutts what do you think?

File: blockio-uring.cabal [hunk 1]
```diff
@@ -73,22 +74,38 @@ benchmark bench
...
+    , vector
 
   pkgconfig-depends: liburing
   other-modules:
     System.IO.BlockIO
     System.IO.BlockIO.URing
     System.IO.BlockIO.URingFFI
 
-  ghc-options:       -threaded -fno-ignore-asserts
+  ghc-options:       -threaded
```
Comment (dcoutts) [inline blockio-uring.cabal:111]: It's true that `ghc-options` like `-fno-ignore-asserts` are not transitive across components. But this use here is not transitive. This test suite does not depend on the library component within the package, it includes the sources for the library directly into the test component (`hs-source-dirs: src`), and thus the `ghc-options` takes effect for the library.
```suggestion
  ghc-options:       -threaded -fno-ignore-asserts
```

Of course for the external test suite you've introduced, that depends on the `blockio-uring` lib and doesn't use `src` in its `hs-source-dirs`, so indeed there `-fno-ignore-asserts` would have no useful effect. But for the internal test suite it does have a useful effect (at least potentially, if we have enough tests to exercise the functionality and provoke any assertions).

**Review by dcoutts — CHANGES_REQUESTED** [review]
I've no problem splitting the tests into internal and external.

Note below that `-fno-ignore-asserts` does have a useful (potentially) effect on the internal lib and so should be retained. Otherwise, good to go.

**Review by dcoutts — APPROVED** [review]
My trivial patch could be squashed.

**Pull request merged**